File: src/Python/414.ThirdMaximumNumber.py

class Solution:
    def thirdMax(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nums = sorted(list(set(nums)))
        return max(nums) if len(nums) < 3 else nums[-3]

if __name__ == '__main__':
    solution = Solution()
    print(solution.thirdMax([2, 2, 3, 1]))
    print(solution.thirdMax([1, 2]))
    print(solution.thirdMax([1, 1, 2]))
    print(solution.thirdMax([3, 2, 1]))
    print(solution.thirdMax([1, 2, 2, 5, 3, 5]))
else:
    pass

File: src/CSharp/414.ThirdMaximumNumber.cs

using System;
using System.Collections.Generic;

namespace PeefyLeetCode.ThirdMax {
public class Solution {
    public int ThirdMax(int[] nums) {
        long first = long.MinValue;
		long secend = long.MinValue;
		long third = long.MinValue;
		for (int i = 0; i < nums.Length; i++) {
			long temp = nums[i];
			if (temp >= first) {
				if (temp != first) {
					third = secend;
					secend = first;
					first = temp;
				}
			} else if (temp >= secend) {
				if (temp != secend) {
					third = secend;
					secend = temp;
				}
			} else if (temp >= third) {
				third = temp;
			}
		}
		if (third == long.MinValue)
			return (int) first;
		return (int) third;
    }
}
}

